### PR TITLE
fix(wallet-lib): separate persistent storage by walletId

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

At the moment, we are storing all the wallets under the same key in the storage persistence. Some chain data is storing seprately from the wallet. In this PR, I separated storage by walletId and chainNetwork. It allows you to switch wallets without unnecessary resync and makes our storage more consistent.

## What was done?

Separated storage of the wallets by walletId (chain, wallets data)

## How Has This Been Tested?

Local tests, CI

## Breaking Changes

Changes persistence storage structure a little bit

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wallet-lib/src/types/WalletStore/methods/exportState.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ function exportState(chainHeight) {
   }
 
   return {
+    walletId,
```
Comment (markin-io) [inline packages/wallet-lib/src/types/WalletStore/methods/exportState.js:14]: I think we don't need it there. Wallet id is already known from the root object. And at importState as well

File: packages/wallet-lib/src/types/Storage/methods/rehydrateState.js [hunk 2]
```diff
@@ -8,47 +8,35 @@ const logger = require('../../../logger');
  * Fetch the state from the persistence adapter
  * @return {Promise<void>}
  */
-const rehydrateState = async function rehydrateState() {
+const rehydrateState = async function rehydrateState(walletId) {
```
Comment (markin-io) [inline packages/wallet-lib/src/types/Storage/methods/rehydrateState.js:11]: Let's leave this function signature intact to not introduce unnecessary changes to the documentation.
You can access a wallet Id with `this.currentWalletId`

**Review by markin-io — CHANGES_REQUESTED** [review]
Good job! Let's do some minor tweaks and we're good to go

**Review by markin-io — DISMISSED** [review]
utACK

**Review by markin-io — APPROVED** [review]

**Comment by thephez** [comment]
This PR has a v0.22.x milestone, but was merged into 0.23-dev. Is there a plan to backport to 0.22 or is the milestone just incorrect?

**Comment by markin-io** [comment]
@thephez thanks. Updated to a correct milestone. 
It will be backported to v0.22 as well

**Comment by shumkov** [comment]
Won't it be a breaking change?

**Comment by markin-io** [comment]
> Won't it be a breaking change?

No, it will wipe old storage and re-sync from the beginning.

**Pull request merged**